### PR TITLE
Adding Pay @ Table and Pre-auth features

### PR DESCRIPTION
**Pull request by metinavc1**

### CODE REVIEW

File: motelpos/MotelPos/Logic/Extensions/toString.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        if (includeTime) {
+            format = "dd/MM/yyyy hh:mm a"
+        } else {
+            format = "dd/MM/yyyy"
+        }
+        return toString(format: format)
+    }
+    
+    func toString(format: String) -> String {
+        let dateFormatter = DateFormatter()
```
Comment (chrishulbert) [inline motelpos/MotelPos/Logic/Extensions/toString.swift:24]: Creating new DateFormatter is really expensive and slow, just a heads up - if this is called iterating over heaps of rows that'd be a problem. If used once per request then no problem.

File: motelpos/MotelPos/View Controllers/Main/MainViewController+PrintStatusActions.swift [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            logMessage(String(format: "# Waiting confirm from POS? %@", String(pairingState.isAwaitingCheckFromPos)))
+            break
+        case .transaction:
+            guard let txState = state.txFlowState else { return }
+            
+            logMessage("### TX PROCESS UPDATE ###")
+            logMessage(String(format: "# %@", txState.displayMessage))
+            logMessage(String(format: "# ID: %@", txState.posRefId))
+            logMessage(String(format: "# Type: %@", txState.type.name))
+            logMessage(String(format: "# Amount: %.2f", Float(txState.amountCents) / 100.0))
```
Comment (chrishulbert) [inline motelpos/MotelPos/View Controllers/Main/MainViewController+PrintStatusActions.swift]: I'd be more comfortable if this were a Double

**Review by chrishulbert — APPROVED** [review]
All looks good, just decide for yourself if my comments are worth worrying about. I'm happy as is

File: motelpos/MotelPos/View Controllers/Main/MainViewController+PrintStatusActions.swift (hunk 2 above)
Comment (metinavc1) [reply motelpos/MotelPos/View Controllers/Main/MainViewController+PrintStatusActions.swift]: it's done

**Pull request merged**